### PR TITLE
[ci skip] document collection_caching.rb

### PR DESCRIPTION
**Pull request by schneems**
I didn't understand what the code does, so I added docs. All methods are private so this is for internal use only.

### CODE REVIEW

File: actionview/lib/action_view/renderer/partial_renderer/collection_caching.rb [hunk 1]
```diff
@@ -14,15 +14,35 @@ module CollectionCaching # :nodoc:
       def cache_collection_render(instrumentation_payload)
         return yield unless @options[:cached]
 
+        # Result is a hash with the key represents the
```
Comment (rafaelfranca) [inline actionview/lib/action_view/renderer/partial_renderer/collection_caching.rb]: Is not better to document this in the method?

Comment (schneems) [reply actionview/lib/action_view/renderer/partial_renderer/collection_caching.rb]: If blame weren't an issue I would change some of the variables to indicate what they are i.e. `keyed_collection_hash`. 

The method return and behavior is actually somewhat easier to understand. 

This one method where I wrote all these docs is essentially where the "magic happens" and I had a hard time piecing together exactly how everything fit together. These are the slightly polished notes that I took when going through that process of understanding this code. Happy to make any changes you think that could aid in comprehension of how all the pieces fit together.

**Review by rafaelfranca — APPROVED** [review]

File: actionview/lib/action_view/renderer/partial_renderer/collection_caching.rb (hunk 1 above)
Comment (rafaelfranca) [reply actionview/lib/action_view/renderer/partial_renderer/collection_caching.rb]: I'm fine with it here too.

**Comment by kaspth** [comment]
Could be authors bias, but I find this documentation to make things harder to understand. Now I first have to understand the text before I can get on with the code. And that’s tough because it references the code so much. My take.

Interspersing the docs throughout breaks the code flow for me and gets in the way. So I’d put the docs outside of the methods.

**Pull request merged**